### PR TITLE
Add support for i18next 23.x and react-i18next 13.x

### PR DESCRIPTION
**Pull request by stevensacks**
Added support for the new major versions of i18next and react-i18next.

### CODE REVIEW

**Review by trishrempel — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -81,10 +81,10 @@
     "eslint-plugin-promise": "^6.1.1",
     "eslint-plugin-testing-library": "^5.9.1",
     "eslint-plugin-unicorn": "^45.0.0",
-    "i18next": "^22.0.6",
+    "i18next": "23.0.2",
```
Comment (trishrempel) [inline package.json:84]: Should these dependencies have carets?

```suggestion
    "i18next": "^23.0.2",
```

**Comment by trishrempel** [comment]
+1 for this PR; I currently can't upgrade `i18next`/`react-i18next` on my Remix project without breaking peer dependencies with `remix-i18next`.

File: package.json (hunk 1 above)
Comment (stevensacks) [reply package.json]: Ah good catch. Fixed.

File: package.json [hunk 2]
```diff
@@ -81,10 +81,10 @@
...
     "prettier": "^2.8.0",
     "react": "^18.2.0",
-    "react-i18next": "^12.0.0",
+    "react-i18next": "13.0.0",
```
[the remark this answers is not in the document]
Comment (stevensacks) [reply package.json]: Fixed.

**Comment by wolthers** [comment]
Anything I can do to help this along?

**Review by sergiodxa — APPROVED** [review]

**Pull request merged**